src/frontend/magnify/src/services/array/array.helper.ts

```diff
@@ -0,0 +1,18 @@
+export class ArrayHelper {
+  public static findElementIndex<T>(array: T[], element: T): number | null {
+    const index = array.findIndex((arrayElement: T) => {
+      return arrayElement == element;
+    });
+    return index >= 0 ? index : null;
+  }
+
+  public static removeItem<T>(array: T[], element: T): T[] {
+    const index = ArrayHelper.findElementIndex(array, element);
+    if (index != null) {
+      array.splice(index, 1);
```

Comment:
In order to prevent side effect, I'll do not use a method which update the array itself.
Furthermore, this method return the array, so as a dev it let me think that the array passed as argument has not been updated.

Reply:
Agreed